plaso/storage/redis/writer.py

```diff
@@ -0,0 +1,357 @@
+      OSError: if the storage writer is closed.
+    """
+    if not self._store:
+      raise IOError('Unable to read from closed storage writer.')
+
+    return None
+
+  def GetSortedEvents(self, time_range=None):
+    """Retrieves the events in increasing chronological order.
+
+    This includes all events written to the storage including those pending
+    being flushed (written) to the storage.
+
+    Args:
+      time_range (Optional[TimeRange]): time range used to filter events
+          that fall in a specific period.
+
+    Returns:
+      generator(EventObject): event generator.
+
+    Raises:
+      IOError: if the storage writer is closed.
+      OSError: if the storage writer is closed.
+    """
+    if not self._store:
+      raise IOError('Unable to read from closed storage writer.')
+
+    return self._store.GetSortedEvents(time_range=time_range)
+
+  def ReadPreprocessingInformation(self, knowledge_base):
+    """Reads preprocessing information.
+
+    Args:
+      knowledge_base (KnowledgeBase): is used to store the preprocessing
+          information.
+
+    Raises:
+      IOError: if the storage format does not support preprocessing information.
+      OSError: if the storage format does not support preprocessing information.
+    """
+    raise IOError('Preprocessing information not supported by the redis store.')
+
+  def RemoveProcessedTaskStorage(self, task):
+    """Removes a processed task storage.
+
+    Args:
+      task (Task): task.
+
+    Raises:
+      IOError: if the storage format does not support removing a task store.
+      OSError: if the storage format does not support removing a task store.
+    """
+    raise IOError(
+        'Removing a task store is not supported by the redis store.')
+
+  def SetSerializersProfiler(self, serializers_profiler):
+    """Sets the serializers profiler.
+
+    Args:
+      serializers_profiler (SerializersProfiler): serializers profiler.
+    """
+    self._serializers_profiler = serializers_profiler
+
+  def SetStorageProfiler(self, storage_profiler):
+    """Sets the storage profiler.
+
+    Args:
+      storage_profiler (StorageProfiler): storage profiler.
+    """
+    self._storage_profiler = storage_profiler
+
+  def WritePreprocessingInformation(self, knowledge_base):
+    """Writes preprocessing information.
+
+    Args:
+      knowledge_base (KnowledgeBase): contains the preprocessing information.
+
+    Raises:
+      IOError: if the storage format does not support preprocessing information.
+      OSError: if the storage format does not support preprocessing information.
+    """
+    raise IOError(
+        'Preprocessing information is not supported by the redis store.')
+
+  def WriteSessionCompletion(self, aborted=False):
+    """Writes session completion information.
+
+    Args:
+      aborted (Optional[bool]): True if the session was aborted.
+
+    Raises:
+      IOError: if the storage type does not support writing a session
```

Comment:
always since the redis ....

Reply:
Nit technically SQLite file store does allow to store session information for task stores at the moment. We likely want to make this more strict seeing this.

Reply:
FYI I'm introduction this check in https://github.com/log2timeline/plaso/pull/2939